### PR TITLE
Do not count calls to 'System._warning()' as violations

### PR DESCRIPTION
**Pull request by mthuurne**
See #302 for details.

### CODE REVIEW

File: pydoctor/model.py [hunk 1]
```diff
@@ -508,7 +508,7 @@ class System:
     def __init__(self, options: Optional[Values] = None):
         self.allobjects: Dict[str, Documentable] = {}
         self.rootobjects: List[Documentable] = []
-        self.warnings: Dict[str, List[Tuple[str, str]]] = {}
+        self.violations: Dict[str, List[Tuple[str, str]]] = {}
```
Comment (adiroiban) [inline pydoctor/model.py:511]: maybe add a docstring with some info about what `violations` are.

I am still not sure how to define them, and what is the difference between `violations` and the remaining `_warnings`.

I would say that a warning like `cannot parse` should end up with a build failure...and not just be ignored.

**Review by adiroiban — APPROVED** [review]
It looks good. Thanks!

I think we can merge it as it is now and later see how to resolve #302  and see which of the remaking warnings are `violations` or `errors`.

I would still like to have pydoctor fail on a non-clean build... and a warning like `cannot parse` should mark the build as "dirty" :)

**Review by adiroiban — APPROVED** [review]
Nice :)

Thanks!

**Comment by mthuurne** [comment]
> I think we can merge it as it is now and later see how to resolve #302 and see which of the remaking warnings are `violations` or `errors`.

Yes, we need to re-do the logging/reporting system at some point, but this should be good enough for making `--warnings-as-errors` usable in the coming release.

**Pull request merged**